Source/WebCore/svg/animation/SMILTime.h

```diff
@@ -34,24 +34,19 @@ namespace WebCore {
 class SMILTime {
 public:
     SMILTime() : m_time(0) { }
-    SMILTime(double time) : m_time(time) { ASSERT(!std::isnan(time)); }
     SMILTime(Seconds time) : m_time(time.value()) { ASSERT(!std::isnan(time.value())); }
-    SMILTime(const SMILTime& o) : m_time(o.m_time) { }
+    SMILTime(double time) : m_time(time) { }
```

Comment:
I think we should have a special private constructor that accepts NaN/inf (SMILTime(double time, AcceptNaNInfValues)), and only the static creation functions for SMILTime should be allowed to pass such values, not the generic one.